src/js/view/controls/tizen/tizen-seekbar.ts

```diff
@@ -0,0 +1,218 @@
+}
+
+export default class TizenSeekbar {
+    _model: ViewModel;
+    _api: PlayerAPI;
+    _slider: TimeSliderWithMixins;
+    el: HTMLElement;
+    currentTime: number;
+    thumbnailContainer: HTMLElement;
+    imageWidth?: number;
+    imageHeight?: number;
+
+    constructor(model: ViewModel, api: PlayerAPI, slider: TimeSlider) {
+        this._model = model;
+        this._api = api;
+        this._slider = slider as TimeSliderWithMixins;
+        this.currentTime = 0;
+
+        const el = document.createElement('div');
+        el.className = 'jw-tizen-seekbar';
+        this.el = el;
+
+        const thumbnailContainer = this.thumbnailContainer = createElement(ThumbnailSeekbarTemplate());
+        this.el.appendChild(thumbnailContainer);
+
+        // Set the thumbnail image width and height in case of sprites (all thumbnails in one image)
+        api.on('firstFrame', () => {
+            if (this._slider.thumbnails && this._slider.thumbnails.length !== 0) {
+                this._setImageDimensions();
+            }
+        });
+    }
+
+    _setImageDimensions(): void {
+        const img = new Image();
+        img.onload = () => {
+            img.onload = null;
+            this.imageWidth = img.width;
+            this.imageHeight = img.height;
+        };
+        img.src = this._slider.chooseThumbnail(0);
+    }
+
+    show(): void {
+        if (this._slider.thumbnails && this._slider.thumbnails.length !== 0) {
+            addClass(this.thumbnailContainer, 'jw-open');
+            this._updateThumbnails(this._model.get('position'), 10);
+        } else {
+            addClass(this._slider.timeTip.el, 'jw-open');
+            this._updateTimeTip(this._model.get('position'));
+        }
+    }
+
+    update(increment: number): void {
+        const currentTime = this.currentTime;
+        const duration = this._model.get('duration');
+        
+        let position = currentTime + increment;
+        position = position < 0 ? 0 : position;
+        position = position > duration ? duration : position;
+
+        if (position !== currentTime) { 
+            if (this._slider.thumbnails && this._slider.thumbnails.length !== 0) {
+                this._updateThumbnails(position, increment);
+            } else {
+                this._updateTimeTip(position);
+            }
+        }
+    }
+
+    _updateTimeTip(position: number): void {
+        const timeTip = this._slider.timeTip;
+        const pct = this._getProgressPercent(position);
+
+        style(timeTip.el, { left: pct + '%' });
+        timeTip.update(timeFormat(Math.round(position)));
+
+        this._updateProgressBar(position);
+        this.currentTime = Math.round(position);
+    }
+
+    _updateThumbnails(position: number, increment: number): void {
+        const thumbnails = this.thumbnailContainer.children;
+        const duration = this._model.get('duration');
+        const positions = [
+            position - (2 * increment),
```

Comment:
This looks like we're absolutely positioning something that should be relative. What couldn't be accomplished via CSS?

Reply:
I can rename this (and I need to actually change the logic). This isn't CSS positioning, it's the timestamps for the thumbnails that will be loaded. 5 are loaded in total, so this is just an array of those 5 timestamps spaced out by the increment.